src/uct/ib/dc/base/dc_ep.c

```diff
@@ -264,6 +261,10 @@ ucs_status_t uct_dc_ep_check_fc(uct_dc_iface_t *iface, uct_dc_ep_t *ep)
                 return status;
             }
             ep->fc.flags |= UCT_DC_EP_FC_FLAG_WAIT_FOR_GRANT;
+            /* Update win to avoid sending another fc request if the message,
```

Comment:
let's add assertion that we don't ask for grant more than once. if we ask for gran while already waiting for grant, the flush logic is wrong.